tests/lib/rules/no-useless-constructor.js

```diff
@@ -47,39 +47,104 @@ ruleTester.run("no-useless-constructor", rule, {
+                    output: "class A {  }"
+                }]
+            }]
         },
         {
             code: "class A extends B { constructor() { super(); } }",
-            errors: [error]
+            errors: [{
+                ...error,
+                suggestions: [{
+                    messageId: "removeConstructor",
+                    output: "class A extends B {  }"
+                }]
+            }]
         },
         {
             code: "class A extends B { constructor(foo){ super(foo); } }",
-            errors: [error]
+            errors: [{
+                ...error,
+                suggestions: [{
+                    messageId: "removeConstructor",
+                    output: "class A extends B {  }"
+                }]
+            }]
         },
         {
             code: "class A extends B { constructor(foo, bar){ super(foo, bar); } }",
-            errors: [error]
+            errors: [{
+                ...error,
+                suggestions: [{
+                    messageId: "removeConstructor",
+                    output: "class A extends B {  }"
+                }]
+            }]
         },
         {
             code: "class A extends B { constructor(...args){ super(...args); } }",
-            errors: [error]
+            errors: [{
+                ...error,
+                suggestions: [{
+                    messageId: "removeConstructor",
+                    output: "class A extends B {  }"
+                }]
+            }]
         },
         {
             code: "class A extends B.C { constructor() { super(...arguments); } }",
-            errors: [error]
+            errors: [{
+                ...error,
+                suggestions: [{
+                    messageId: "removeConstructor",
+                    output: "class A extends B.C {  }"
+                }]
+            }]
         },
         {
             code: "class A extends B { constructor(a, b, ...c) { super(...arguments); } }",
-            errors: [error]
+            errors: [{
+                ...error,
+                suggestions: [{
+                    messageId: "removeConstructor",
+                    output: "class A extends B {  }"
+                }]
+            }]
         },
         {
             code: "class A extends B { constructor(a, b, ...c) { super(a, b, ...c); } }",
-            errors: [error]
+            errors: [{
+                ...error,
+                suggestions: [{
+                    messageId: "removeConstructor",
+                    output: "class A extends B {  }"
+                }]
+            }]
+        },
+        {
+            code: "class A {\nfoo = 5\nconstructor() {}\nbar() {}\n}",
+            languageOptions: { ecmaVersion: 2022 },
+            errors: [{
+                ...error,
+                suggestions: [{
+                    messageId: "removeConstructor",
+                    output: "class A {\nfoo = 5;\n\nbar() {}\n}"
```

Comment:
removed this test case

Reply:
> OOC are there any resources available that I could learn from to know when a semi colon should be inserted?

Nothing specific to ESLint that I know of. The rules for automatic semicolon insertion are part of the JavaScript specification. MDN has [a chapter](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Lexical_grammar#automatic_semicolon_insertion) about it with some helpful examples.
In practice one will want to look at the parser output in most cases to be sure if a semicolon was inserted or not.

Generally in ESLint when a suggestion or autofix removes a definition, we take care not to change the statements before or after it.
This could happen for example when there is an implicit semicolon before the definition, and if the definition is removed, that semicolon would no longer be inserted automatically.

A notorious case is when the preceding node is part of an expression (on another line, without a trailing semicolon), and the token that follows is an operator that could continue the previous expression. This includes many common tokens that change their meaning depending on the context, for example:
* `+`, `-`: can be unary or binary operators
* `/` can introduce a regular expression or indicate a division
* `(` used to group an expression, the argument list of an arrow function, or to introduce a function call
* `[` for array literals, computed member names, or for member access
* `` ` `` for untagged or tagged templates

In the body of a class, due to syntax requirements, the only one of these tokens that can appear at the beginning of a statement is `[`, and for that case we have a function ready to use; but in other cases, detecting if a semicolon should be inserted needs work.